controllers/azuremanagedcontrolplane_controller_test.go

```diff
@@ -105,6 +106,8 @@ func TestAzureManagedControlPlaneReconcilePaused(t *testing.T) {
 		infrav1.AddToScheme,
 		asoresourcesv1.AddToScheme,
 		asocontainerservicev1.AddToScheme,
+		corev1.AddToScheme,
+		aadpodv1.AddToScheme,
```

Comment:
I assume this is needed for the test to pass? But ideally we wouldn't be adding usage of AAD Pod ID; see #4249.

Reply:
Removing it causes this error:
```
failed to create scope: failed to configure azure settings and credentials for Identity: failed to create copied AzureIdentity paused-kqvhtglxrf-default-fake-identity in capz-system: no kind is registered for the type v1.AzureIdentity in scheme "pkg/runtime/scheme.go:100"
```
This error happens on this [line](https://github.com/willie-yao/cluster-api-provider-azure/blob/remove-setcredentials/azure/scope/identity.go#L297). Is this something you'd remove in your PR?

Reply:
Yes, I can address that when fixing #4249.